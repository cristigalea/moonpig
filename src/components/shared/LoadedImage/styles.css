.image__wrapper {
    position: relative;
    min-width: 270px;
    min-height: 190px;
    display: flex;
    align-items: flex-end;
    justify-content: center;
}

.image__content {
    max-height: 340px;
    max-width: 270px;
}

.image__loader {
    position: absolute;
    bottom: 10%;
    left: 40%;
    width: 50px;
}
.image__error {
    position: absolute;
    bottom: 0;
    left: 0;
    width: 280px;
}