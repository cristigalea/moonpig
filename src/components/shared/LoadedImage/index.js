import React, {Component} from 'react';
import PropTypes from 'prop-types';

import './styles.css';

export default class LoadedImage extends Component {

    static propTypes = {
        url: PropTypes.string
    };

    state = {
        isError: false,
        isLoaded: false
    };

    onLoadImage = () => {
        this.setState({
            isError: false,
            isLoaded: true
        });
    };

    onErrorImage = () => {
        this.setState({
            isError: true
        });
    };

    render() {
        const { url } = this.props;
        const { isLoaded, isError } = this.state;
        const imgStyle = { opacity: isLoaded && !isError ? 1 : 0 };

        return (
            <div className="image__wrapper">
                <img className="image__content" style={imgStyle} src={url} onLoad={this.onLoadImage} onError={this.onErrorImage} />
                {!isLoaded && !isError && <img className="image__loader" src="https://dosrgfkou9o2m.cloudfront.net/cdn/3.0.0.16195/responsivemasterpage/img/3-dots@2x.gif"/>}
                {isError && <img className="image__error" src="error-image.png"/>}
            </div>
        )
    }
}