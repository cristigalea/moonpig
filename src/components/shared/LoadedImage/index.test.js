import React from 'react';
import {shallow} from 'enzyme';

import LoadedImage from '.';

describe('<LoadedImage />', () => {
    it('should match the snapshot', () => {
        const component = shallow(<LoadedImage url="https://cdn.britannica.com/s:700x450/01/115901-004-AC4BD3CC.jpg" />);
        expect(component).toMatchSnapshot();
    });

    it('should finish loading', () => {
        const component = shallow(<LoadedImage url="https://cdn.britannica.com/s:700x450/01/115901-004-AC4BD3CC.jpg" />);
        const handler = jest.spyOn(component.instance(), 'onLoadImage');
        expect(component.state().isLoaded).toBe(false);
        expect(component.state().isError).toBe(false);
        setTimeout(500, () => {
            expect(handler.mock.calls.length).toBe(1);
            expect(component.state().isLoaded).toBe(true);
        })
    });
});