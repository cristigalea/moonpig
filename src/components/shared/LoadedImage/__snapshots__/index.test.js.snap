// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LoadedImage /> should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <LoadedImage
    url="https://cdn.britannica.com/s:700x450/01/115901-004-AC4BD3CC.jpg"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <img
          className="image__content"
          onError={[Function]}
          onLoad={[Function]}
          src="https://cdn.britannica.com/s:700x450/01/115901-004-AC4BD3CC.jpg"
          style={
            Object {
              "opacity": 0,
            }
          }
        />,
        <img
          className="image__loader"
          src="https://dosrgfkou9o2m.cloudfront.net/cdn/3.0.0.16195/responsivemasterpage/img/3-dots@2x.gif"
        />,
        false,
      ],
      "className": "image__wrapper",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "image__content",
          "onError": [Function],
          "onLoad": [Function],
          "src": "https://cdn.britannica.com/s:700x450/01/115901-004-AC4BD3CC.jpg",
          "style": Object {
            "opacity": 0,
          },
        },
        "ref": null,
        "rendered": null,
        "type": "img",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "image__loader",
          "src": "https://dosrgfkou9o2m.cloudfront.net/cdn/3.0.0.16195/responsivemasterpage/img/3-dots@2x.gif",
        },
        "ref": null,
        "rendered": null,
        "type": "img",
      },
      false,
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <img
            className="image__content"
            onError={[Function]}
            onLoad={[Function]}
            src="https://cdn.britannica.com/s:700x450/01/115901-004-AC4BD3CC.jpg"
            style={
              Object {
                "opacity": 0,
              }
            }
          />,
          <img
            className="image__loader"
            src="https://dosrgfkou9o2m.cloudfront.net/cdn/3.0.0.16195/responsivemasterpage/img/3-dots@2x.gif"
          />,
          false,
        ],
        "className": "image__wrapper",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "image__content",
            "onError": [Function],
            "onLoad": [Function],
            "src": "https://cdn.britannica.com/s:700x450/01/115901-004-AC4BD3CC.jpg",
            "style": Object {
              "opacity": 0,
            },
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "image__loader",
            "src": "https://dosrgfkou9o2m.cloudfront.net/cdn/3.0.0.16195/responsivemasterpage/img/3-dots@2x.gif",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        false,
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
